File: lesson_5/code/mandelbrot_set.m
    % Parameters
    points = 2000;
    i = 100;  
    
    % Create a grid of complex numbers
    x = linspace(-2, 1, points);
    y = linspace(-1.5, 1.5, points);
    [X, Y] = meshgrid(x, y);
    C = X + 1i * Y;
    Z = zeros(size(C));
    divergence = zeros(size(C));


    % Set up the figure
    figure('Position', [100, 100, 1000, 1000]); 
    colormap("gray")
    axis equal;
    
    % Mandelbrot calculation
    for n = 1:i
        Z = Z.^2 + C;
        divergence(abs(Z) > 2) = 1;

        figure(1)
        imagesc(x, y, divergence);
    end
    

File: lesson_5/code/mandelbrot_set.py
import matplotlib.pyplot as plt
import numpy as np


points = 2000
iterations = 100
x = np.linspace(-2, 1, points)
y = np.linspace(-1.5, 1.5, points)
xx, yy = np.meshgrid(x, y)
c = xx + 1j * yy
z = np.zeros_like(c)
divergence = np.zeros(c.shape, dtype=int)

plt.figure(figsize=(10, 10))
for _ in range(iterations):
    z = z ** 2 + c
    divergence[np.abs(z) > 2] = 1
    plt.clf()
    plt.imshow(divergence, extent=[x.min(), x.max(), y.min(), y.max()], origin="lower", cmap="gray")
    plt.pause(0.001)

plt.show()
